lib/src/app_credentials.dart

```diff
@@ -41,6 +41,14 @@ class Credentials {
   Credentials.anonymous()
       : _handle = realmCore.createAppCredentialsAnonymous(),
         provider = AuthProvider.anonymous;
+
+  /// Returns a [Credentials] object that can be used to authenticate a user with their email and password.
+  /// A user can login with email and password only after they have registered their account and verified their
+  /// email.
+  /// [Email/Password Authentication Docs](https://docs.mongodb.com/realm/authentication/email-password)
+  Credentials.emailPassword(String email, String password)
+      : _handle = realmCore.createAppCredentialsEmailPassword(email, password),
```

Comment:
This ties into @nirinchev's proposal, that we should not create the handle until actually needed. If we do agree on that then this needs to change somewhat.. 

Fx. we could make a subclass `EmailAndPassworsCredentials` that has two final fields (email & password) and have `_handle` be late with same initializer as here.

But again.. depends on what stance we take on @nirinchev 's PR..

Reply:
To untangle the PR jam, I think we should merge as is. We can always revisit

Reply:
Yes, we can change it later.